src/mlpack/core/data/split_data.hpp

```diff
@@ -171,17 +173,19 @@ void Split(const arma::Mat<T>& input,
  * @return std::tuple containing trainData (arma::Mat<T>), testData
  *      (arma::Mat<T>), trainLabel (arma::Row<U>), and testLabel (arma::Row<U>).
  */
-template<typename T, typename U>
-std::tuple<arma::Mat<T>, arma::Mat<T>, arma::Row<U>, arma::Row<U>>
+template <typename T, class LabelType,
+          class = std::enable_if_t<arma::is_Row<LabelType>::value ||
+                           arma::is_Mat_only<LabelType>::value>>
+std::tuple<arma::Mat<T>, arma::Mat<T>, LabelType, LabelType>
 Split(const arma::Mat<T>& input,
-      const arma::Row<U>& inputLabel,
+      const LabelType& inputLabel,
```

Comment:
Why did you choose to change `arma::Row<U>` to `LabelType`?  Since you're using the restrictor `arma::is_Row<LabelType>::value` anyway, it doesn't seem like there would be any difference?

Reply:
We are expecting two types of inputLabel ```Mat``` and ```row``` hence I made it as ```LabelType``` to accommodate for both.

Reply:
Ahh, understood, thanks for the clarification.  I would consider the name `LabelsType` though; it's pedantic, but technically the "label type" would be the element type held inside of `inputLabels`, whereas the "labels type" would be the type of the container holding the labels.